### PR TITLE
Implement ModalMechanicalSimulation results

### PR DESCRIPTION
**Pull request by PProfizi**
Implement result extraction APIs for ModalMechanicalSimulation. 

# ModalMechanicalSimulation methods 

(checked: tested, italic: implemented)

- [x] _displacement_
- [x] _stress, stress_elemental, stress_nodal_
- [x] _stress_principal, stress_principal_nodal, stress_principal_elemental_
- [x] _stress_eqv_von_mises, stress_eqv_von_mises_elemental, stress_eqv_von_mises_nodal_
- [x] _elastic_strain, elastic_strain_nodal, elastic_strain_elemental_
- [x] _elastic_strain_principal, elastic_strain_principal_nodal, elastic_strain_principal_elemental_
- [ ] _elastic_strain_eqv_von_mises, elastic_strain_eqv_von_mises_elemental, elastic_strain_eqv_von_mises_nodal_
- [ ] elastic_strain_energy_density
- [ ] _plastic_state_variable_
- [ ] _plastic_strain, plastic_strain_nodal, plastic_strain_elemental_
- [ ] _plastic_strain_principal, plastic_strain_principal_nodal, plastic_strain_principal_elemental_
- [ ] _plastic_strain_eqv, plastic_strain_eqv_elemental, plastic_strain_eqv_nodal_
- [ ] plastic_strain_energy_density
- [x] _reaction_force_
- [x] _element_nodal_forces, element_nodal_forces_nodal, element_nodal_forces_elemental_
- [ ] _nodal_force_
- [ ] _nodal_moment_
- [ ] _thickness_
- [ ] _element_centroids_
- [ ] _element_orientations_
- [ ] _elemental_mass_
- [x] _elemental_volume_
- [ ] _hydrostatic pressure_

### CODE REVIEW

File: src/ansys/dpf/post/modal_mechanical_simulation.py [hunk 1]
```diff
@@ -0,0 +1,4053 @@
...
+            named_selections:
+                Named selection or list of named selections to get results for.
+
+        Returns
+        -------
+            Returns a :class:`ansys.dpf.post.data_object.DataObject` instance.
+
+        """
+        # Build the targeted time scoping
+        time_scoping = self._build_time_freq_scoping(
```
Comment (cbellot000) [inline src/ansys/dpf/post/modal_mechanical_simulation.py:78]: @PProfizi sorry to see that late, but didn't we say we would use the selection object for that? To centralize the code implementation on it

Comment (cbellot000) [reply src/ansys/dpf/post/modal_mechanical_simulation.py:78]: also be cause we need to work withh workflows and it's what the selection does

File: src/ansys/dpf/post/modal_mechanical_simulation.py [hunk 2]
```diff
@@ -0,0 +1,4053 @@
...
+        )
+
+    def stress_elemental(
+        self,
+        element_ids: Union[List[int], None] = None,
+        frequencies: Union[float, List[float], None] = None,
+        components: Union[str, List[str], int, List[int], None] = None,
+        set_ids: Union[int, List[int], None] = None,
+        all_sets: bool = False,
+        load_steps: Union[
```
Comment (cbellot000) [inline src/ansys/dpf/post/modal_mechanical_simulation.py:337]: @PProfizi do we have load steps in modal?

File: src/ansys/dpf/post/modal_mechanical_simulation.py [hunk 3]
```diff
@@ -0,0 +1,4053 @@
...
+            int, List[int], Tuple[int, Union[int, List[int]]], None
+        ] = None,
+        named_selections: Union[List[str], str, None] = None,
+        selection: Union[Selection, None] = None,
```
Comment (cbellot000) [inline src/ansys/dpf/post/modal_mechanical_simulation.py:847]: @PProfizi after yesterday's discussion, will you add location here?

Comment (PProfizi) [reply src/ansys/dpf/post/modal_mechanical_simulation.py:847]: @cbellot000 yes, I will add it to all types of simulation.

File: src/ansys/dpf/post/modal_mechanical_simulation.py (hunk 2 above)
Comment (PProfizi) [reply src/ansys/dpf/post/modal_mechanical_simulation.py:337]: @cbellot000 I'll expose loadsteps and substeps as "modes".

File: src/ansys/dpf/post/modal_mechanical_simulation.py (hunk 1 above)
Comment (PProfizi) [reply src/ansys/dpf/post/modal_mechanical_simulation.py]: @cbellot000 you are right, this first version was using the Selection API to create one on-the-fly and use its workflow.
I've added this for ModalMechanicalSimulation.

**Review by PProfizi — COMMENTED** [review]
Will change the behavior when asking for time values in a later PR.
There are design questions regarding this, especially concerning errors raised when a value is not available. If we go through the operator's time values selection/interpolation, only a server error is raised and only at evaluation.

**Review by cbellot000 — APPROVED** [review]

**Pull request merged**